zebra-chain/src/error.rs

```diff
@@ -57,17 +57,47 @@ pub enum AddressError {
 }
 
 /// `zebra-chain`'s errors
-#[derive(Error, Debug)]
+#[derive(Clone, Error, Debug)]
 pub enum Error {
     /// Invalid consensus branch ID.
     #[error("invalid consensus branch id")]
     InvalidConsensusBranchId,
 
     /// Zebra's type could not be converted to its librustzcash equivalent.
-    #[error("Zebra's type could not be converted to its librustzcash equivalent: ")]
-    Conversion(#[from] io::Error),
+    #[error("Zebra's type could not be converted to its librustzcash equivalent: {0}")]
+    Conversion(#[from] Arc<io::Error>),
 
     /// The transaction is missing a network upgrade.
     #[error("the transaction is missing a network upgrade")]
     MissingNetworkUpgrade,
 }
+
+/// Allow converting `io::Error` to `Error`; we need this since we
+/// use `Arc<io::Error>` in `Error::Conversion`.
+impl From<io::Error> for Error {
+    fn from(value: io::Error) -> Self {
+        Arc::new(value).into()
+    }
+}
+
+// We need to implement this manually because io::Error does not implement
+// PartialEq.
+impl PartialEq for Error {
+    fn eq(&self, other: &Self) -> bool {
+        match self {
+            Error::InvalidConsensusBranchId => matches!(other, Error::InvalidConsensusBranchId),
+            Error::Conversion(e) => {
+                if let Error::Conversion(o) = other {
+                    // Not perfect, but good enough for testing, which
+                    // is the main purpose for our usage of PartialEq for errors
+                    e.to_string() == o.to_string()
+                } else {
+                    false
+                }
+            }
+            Error::MissingNetworkUpgrade => matches!(other, Error::MissingNetworkUpgrade),
+        }
+    }
+}
```

Comment:
We should implement this for the error type in `zebra-script` instead and say it's partially equal as long as the enum variant matches. I think we should avoid implementing `Eq` for that too.

Reply:
> Why do you think that?

I think it's okay because that was the behaviour before, but, it's not ideal.